app/celery/tasks.py

```diff
@@ -113,7 +115,10 @@ def create_pdf_for_templated_letter(encrypted_letter_data):
     with current_app.test_request_context(''):
         html = HTML(string=str(template))
 
-    pdf = BytesIO(html.write_pdf())
+    try:
+        pdf = BytesIO(html.write_pdf())
+    except werkzeug.exceptions.BadGateway as exc:
+        self.retry(exc=exc)
```

Comment:
I think you need to add `queue` here as a kwarg (I looked in the docs and couldn't see which queue would be used if that's not specified).

Reply:
We paired on this and confirmed the `queue` kwarg is necessary, which is strange because [the docs](https://docs.celeryproject.org/en/3.1/userguide/tasks.html#retrying) for v3 celery explicitly say: "it will take care to make sure the message is delivered to the same queue as the originating task", but the code doesn't clearly match up to that ([here](https://github.com/celery/celery/blob/6cf4f40fc58fe8585721f4df95a1c77d25106dbf/celery/app/task.py#L600)). There's also [this issue](https://github.com/celery/celery/issues/3622), suggesting a similar problem.

Anyway, I've amended the commit and specified the queue as "SANITISE_LETTERS", which is the one the API app creates instances of this task on ([here](https://github.com/alphagov/notifications-api/blob/master/app/celery/letters_pdf_tasks.py#L89)). We _could_ have a special "RETRY" queue like the API uses, but that seems overkill for something that we expect to be infrequent and for which we only retry for a short period.

(Also tested it works as expected.)